pallet-chainlink-feed/src/lib.rs

```diff
@@ -38,10 +38,18 @@ pub mod pallet {
 	use sp_std::prelude::*;
 
 	use crate::{
+		default_weights::SubmitWeight,
 		traits::OnAnswerHandler,
 		utils::{median, with_transaction_result},
 	};
 
+	/// Configuration for submiting paysfee
+	pub enum SubmitPaysFee {
+		Yes,
+		No,
+		Auto,
+	}
```

Comment:
I pictured this a little differently.
i think we should only two options: for submitting the normal fee applies as already implemented in `Weights` or nothing if it is a valid submitter.
So this enum ideally should have two variants, `Always` and something like `FreeForValidRound`. 
If a submitter submits a invalid round he should need to pay for the usual weight.